File: ruby/bob/bob.rb
class Bob
  def self.hey(remark)
    if (remark == remark.upcase) && (remark.count("a-zA-Z") > 0)
      return "Whoa, chill out!"
    elsif remark.rstrip[-1] == "?"
      return "Sure."
    elsif remark.strip.length == 0
      return 'Fine. Be that way!'
    else
      return 'Whatever.'
    end
  end

end


module BookKeeping
  VERSION = 1
end

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (input.length === 0 || input[0] === " ") {
    return 'Fine. Be that way!'
  } else if (input === input.toUpperCase() && input == input.toLowerCase() && input[0] !== '4') {
    return 'Whatever.'
  } else if (input === input.toUpperCase() && input[0] !== '4') {
    return 'Whoa, chill out!';
  } else if (input.slice(-1) === '?') {
    return 'Sure.'
  }
  return 'Whatever.'
};

module.exports = Bob;
